servicetest/testframework/testhelper.py

```diff
@@ -206,12 +207,43 @@ def result(self):
             file_content = fh.readline()
         return int(file_content)
 
+class CoreDumpHelper(Helper):
+    """ Helper for checking for core dumps
+    """
+
+    def __init__(self, file_base_path):
+        Helper.__init__(self, file_base_path)
+
+    def checkPattern(self, expected_pattern):
+        with open("/proc/sys/kernel/core_pattern", "r") as core_pattern:
+            with open(self._base_path + "/core_pattern", "w") as fh:
+                pattern = core_pattern.read()
+                fh.write(pattern)
+                LOG(pattern)
+
+    def patternResult(self):
+        with open(self._base_path + "/core_pattern", "r") as fh:
+            return fh.read()
+
+    def assertLimit(self):
+        (soft, hard) = resource.getrlimit(resource.RLIMIT_CORE)
+        with open(self._base_path + "/core_limits", "w") as fh:
+            fh.write(str(soft))
+
+    def limitResult(self):
+        with open(self._base_path + "/core_limits", "r") as fh:
+            return fh.read()
+
+    def waitForSignal(self):
+        while True:
+            time.sleep(0.25)
```

Comment:
Would be nice with more comments I guess... you can read yourself to this stuff from the code I guess though.